### PR TITLE
not using s2i deployment dir

### LINKED ISSUE #1471: Use subPath `ODO_S2I_DEPLOYMENT_DIR`
We are using `ODO_S2I_DEPLOYMENT_DIR` to backup artifacts that were built in java based s2i builder image and then restore it the run script.

We can completely avoid this step, and instead, just add volume to `ODO_S2I_DEPLOYMENT_DIR`. To avoid creating another PVC, we can reuse the same one that we already use for app-root, and just use directory in the volume.

Something like this?
```yaml
volumeMounts:
  - name: app-root
    mountPath: /opt/app-root
    subPath: app-root
  - name: deployment-dir
    mountPath: <$ODO_S2I_DEPLOYMENT_DIR>
    subPath: deployment-dir
```

### PR DESCRIPTION
**Pull request by girishramnani**
as now we are mounting s2i deployment dir as a persistent volume we dont need to back it up

fixes https://github.com/openshift/odo/issues/1471

### CODE REVIEW

**Comment by cdrage** [comment]
Seems like tests aren't passing but that's because of #24 , once that's merged let's re-run the tests and see what happens.

**Comment by kadel** [comment]
/lgtm

this is a good cleanup of the code that won't be needed once  openshift/odo#1471 is merged

**Comment by cdrage** [comment]
@kadel I think you mean once https://github.com/openshift/odo/pull/1840 is merged? Would this PR be blocked until https://github.com/openshift/odo/pull/1840 is in?

**Comment by kadel** [comment]
> @kadel I think you mean once [openshift/odo#1840](https://github.com/openshift/odo/pull/1840) is merged? Would this PR be blocked until [openshift/odo#1840](https://github.com/openshift/odo/pull/1840) is in?

Nope. 
 - This should be merged first.
 - Then we release the new version of the image
 - Then in openshift/odo#1840 the `defaultBootstrapperImage` needs to be updated to the new image version
 
Only then we can verify that openshift/odo#1840 works properly

**Review by cdrage — APPROVED** [review]
I don't see any issues with the PR!

/approve

**Pull request merged**
